add ipaddress, pool and subnet datasources

tsenay opened the pull request:
Hi, here are the datasources that we spoked about

Commit:
add ipaddress, pool and subnet datasources

alexissavin commented on solidserver/data_source_ip_address.go:
```diff
@@ -0,0 +1,146 @@
+	s := meta.(*SOLIDserver)
+
+	parameters := url.Values{}
+	parameters.Add("ip_id", d.Id())
```
Using the internal ID to find the IP address seems a bit disturbing. This could be done using the Space and IP address itself, in order to retrieve both the ID and meta-data. Would this make sense to you ?

tsenay replied:
The goal of datasource in our case is to get the IP name after created it. We took the shortest path. For a generic usage of datasource, it looks to be a better way to use space and IP

alexissavin commented on solidserver/data_source_ip_address.go:
```diff
@@ -0,0 +1,146 @@
+		Read: dataSourceipaddressRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
```
There is no need to create a field for the ID, there is one by default that can be set with "d.SetId(<ID>)"

tsenay replied:
assumed that's the input parameter, we should declare it ?

alexissavin replied:
I will run some test to validate this point once merged in the branch. Thanks for the clarification.

alexissavin commented on solidserver/data_source_ip_address.go:
```diff
@@ -0,0 +1,146 @@
+				Description: "The class associated to the IP address.",
+				Computed:    true,
+			},
+			"cidr": {
```
By CIDR, do you mean the prefix-size ? If so, I would suggest to name this field "prefix-size" as I plan to to add both the netmask and prefix (CIDR notation) to the data-source and resource.

tsenay replied:
right, and maybe call it "prefix-length" to be precise ?

alexissavin commented on solidserver/data_source_ip_pool.go:
```diff
@@ -0,0 +1,128 @@
+				Description: "The size of the pool.",
+				Computed:    true,
+			},
+			"cidr": {
```
By CIDR, do you mean the prefix-size ? If so, I would suggest to name this field "prefix-size" for consistency.

tsenay replied:
I mean prefix length...

alexissavin reviewed (changes requested):
Hello,

Many thanks for your contribution.

I made some comment in the Pull Request and ill wait for your feedback before merging the code.

Kind regards

alexissavin committed:
Update data_source_ip_address.go

Suggesting reading the IP information from space name and IP address instead of the internal ID - To be discussed, import by ID makes sense but not sure an ID parameter is needed.

alexissavin committed:
Update data_source_ip_pool.go

Just renaming "cidr" into "prefix_size" for consistency within the provider

alexissavin committed:
Merge branch 'master' into feature_datasources

alexissavin commented:
@tsenay,  if I may ask, I do see something strange in this PR:

The prefix_size is computed from the pool_size parameter which is not reflecting the size of the subnet the pool belongs to.
'''
cidrFloat, _ := strconv.ParseFloat(buf[0]["pool_size"].(string), 64)
d.Set("prefix_size", fmt.Sprintf("%d", int(32-math.Round(math.Log(cidrFloat)+2))))
'''
What is the information you are looking for exactly ?
I can compute the prefix and prefix size if needed.

tsenay commented:
The prefix size is used to configure a network card, so we need the size of the subnet

alexissavin commented:
Thanks for your feedback.

Latest commit ea23d29cb50a70b8c7d0b4aa21e93129c5251315 should behave as you expect.
You will also have access to IP Pool resource.

Please let me know if this is fine by you.
